src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,82 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_SIZE = 16;
+    private static final double RESIZE_FACTOR = 0.75;
+    private Node<K, V>[] storage;
+    private int size;
+
+    public MyHashMap() {
+        storage = (Node<K, V>[]) new Node[DEFAULT_SIZE];
+    }
 
     @Override
-    public void put(K key, V value) {
+    public int getSize() {
+        return size;
+    }
 
+    @Override
+    public void put(K key, V value) {
+        Node<K, V> existingNode = getNodeByKey(key);
+        if (existingNode != null) {
+            existingNode.value = value;
+            return;
+        }
+        ensureCapacity();
+        insertNode(new Node<>(key, value));
+        size++;
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        final Node<K, V> existingNode = getNodeByKey(key);
+        return existingNode == null ? null : existingNode.value;
     }
 
-    @Override
-    public int getSize() {
-        return 0;
+    private Node<K, V> getNodeByKey(K key) {
+        Node<K, V> current = storage[getBucketIndex(key)];
+        do {
+            if (current == null || Objects.equals(current.key, key)) {
+                return current;
+            }
+            current = current.next;
+        } while (true);
+    }
+
+    private int getBucketIndex(K key) {
+        return key == null ? 0 : Math.abs(key.hashCode()) % storage.length;
+    }
+
+    private void ensureCapacity() {
+        if (size / (double) storage.length > RESIZE_FACTOR) {
```

Comment:
you can improve you solution.
```suggestion
        if (size > RESIZE_FACTOR * storage.length) {
```
And you don't need to cast